### ISSUE TITLE
have collisoin detection？

### ISSUE THREAD

**Issue by meteorshowers** [aaronwalsman/ltron#5]
have collisoin detection？

**Comment by aaronwalsman** [comment]
Yes, we do support a form of collision detection.  See ltron/geometry/collision.py.  If you set the `collision_checker=True` argument when building a BrickScene, it will build a collision checker for you automatically.  You can then use `scene.check_collision` and `scene.check_snap_collision`.  If you have any information about your use case, I'd be happy to give you some pointers on how to make it work.

**Comment by meteorshowers** [comment]
Thanks.
I can run itron on linux, but do not know how to get the connectivity structure with a graph-based representation in the workshop paper.

**Comment by aaronwalsman** [comment]
Hello!  Sorry, the API has changed a little bit since the workshop paper.  Here's how to get the connectivity graph:

First get a `BrickScene` object that contains the scene information, something like this:
```
from ltron.bricks.brick_scene import BrickScene
scene = BrickScene(track_snaps=True)
scene.import_ldraw('./Black Racer.mpd')
```

Once you have the scene, you can get the connectivity structure using the `.get_assembly()` method:
```
assembly = scene.get_assembly()
```

The `assembly` variable is a dictionary containing `shape`, `color`, `pose` and `edges` keys with the following structure:

The `shape` key is an array of integers describing the class label of the brick shape for all bricks in the scene.  Note that the class labels start at 1, and that a value of 0 indicates no brick present at that index, and is used for padding purposes in batches.

The `color` key is also an array of integers describing the class label of the brick color for all bricks in the scene.  These class labels also start at 1.

The `pose` key is an array of 4x4 transformation matrices for each brick.

Finally the `edges` key defines a list of connections between bricks.  The `edges` array is 4xN, where the first row is the index of the first brick, the second row is the index of the second brick, the third row is the snap ID of the connection point on the first brick and the fourth row is the snap ID of the connection point on the second brick.

If you are doing this using the gym interface, there are easy ways to get at the scene object used by the gym environment, and the current assembly.  Let me know if this is the case and I can send you more details here.

I hope this is helpful.  If you have more questions or want to describe your application, I'd be more than happy to give you some guidance on how to best achieve what you are looking for!
